### PR TITLE
Change convoluted url and name in windowed presentation

### PR DESCRIPTION
**Pull request by NickWingate**
Open an empty popup with a blank url, then set the content of the popup to be our slideURL wrapped inside an iframe with a user friendly title.
Focus the iframe on load so that arrowkey navigation works and forward the eventlisterner of the iframe to the parent window to watch for escape key calls

Change-Id: Ie1063095db51119fd45453b8b4b180fcacb6d7c1

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/map/handler/Map.SlideShow.js [hunk 1]
```diff
@@ -148,7 +148,7 @@ L.Map.SlideShow = L.Handler.extend({
 		// Windowed Presentation
 		if (this._presentInWindow) {
 
-			var popupTitle = "Windowed Presentation: " + this._map['wopi'].BaseFileName;
+			var popupTitle = _("Windowed Presentation: ") + this._map['wopi'].BaseFileName;
```
Comment (pedropintosilva) [inline browser/src/map/handler/Map.SlideShow.js:151]: Tested on FF and chrome, works

File: browser/src/map/handler/Map.SlideShow.js [hunk 2]
```diff
@@ -190,6 +201,44 @@ L.Map.SlideShow = L.Handler.extend({
...
+				}
+				iframe {
+					width: 100%;
+					height: 100%;
+					border: none;
+				}
+			</style>
+		</head>
+		<body>
+			<iframe src="${slideURL}"></iframe>
```
Comment (pedropintosilva) [inline browser/src/map/handler/Map.SlideShow.js:236]: The "url mask" works on chrome but not firefox which seems odd :

image

**Comment by pedropintosilva** [comment]
didn't test the Impress: Present in window: Next/Prev slide key scroll window (HiDPI screen) yet

**Comment by pedropintosilva** [comment]
Thanks @NickWingate ! this is awesome news, I have added some comments from my tests above I hope it helps.

File: browser/src/map/handler/Map.SlideShow.js (hunk 2 above)
Comment (NickWingate) [reply browser/src/map/handler/Map.SlideShow.js:236]: Firefox has automatic url navigation when we use `document.write(...)` and changes the url to the parent window. document.write may also cause additional problems (see https://developer.mozilla.org/en-US/docs/Web/API/Document/write) Solution is to use `document.documentElement.innerHTML = ...` changing only the innerHTML of the popup.

**Review by pedropintosilva — APPROVED** [review]

**Comment by pedropintosilva** [comment]
I have rebased and forced push as there were many cypress changes that were meanwhile merged on master

File: browser/src/map/handler/Map.SlideShow.js [hunk 3]
```diff
@@ -147,16 +147,27 @@ L.Map.SlideShow = L.Handler.extend({
 	_startPlaying: function() {
 		// Windowed Presentation
 		if (this._presentInWindow) {
-			this._slideShowWindowProxy = window.open(this._slideURL, '_blank', 'popup'); // do we need to set this to null when closed or is that already done?
+
+			var popupTitle = "Windowed Presentation: " + this._map['wopi'].BaseFileName;
```
Comment (eszkadev) [inline browser/src/map/handler/Map.SlideShow.js]: This has to be translated string. Please use `_('....')`.
Also everywhere we used `'` for strings here. Can we be consistent inside single file? (somehow we have enabled `"` in master now)

File: browser/src/map/handler/Map.SlideShow.js [hunk 4]
```diff
@@ -147,16 +147,27 @@ L.Map.SlideShow = L.Handler.extend({
...
+					_('Presentation was blocked. Please allow pop-ups in your browser. This lets slide shows to be displayed in separated windows, allowing for easy screen sharing.'), '',
+					_('OK'), null, false);
 			}
 
+			this._slideShowWindowProxy.document.write(htmlContent)
+			this._slideShowWindowProxy.document.close()
 			this._slideShowWindowProxy.focus();
-			this._slideShowWindowProxy.addEventListener('keydown', this._onCloseSlideWindow.bind(this));
+
+			this._slideShowWindowProxy.onload = this._handleSlideWindowLoaded();
```
[the remark this answers is not in the document]
Comment (eszkadev) [reply browser/src/map/handler/Map.SlideShow.js:167]: `... = this._handleSlideWindowLoaded.bind(this)`

**Review by eszkadev — CHANGES_REQUESTED** [review]
added some comments

File: browser/src/map/handler/Map.SlideShow.js [hunk 5]
```diff
@@ -190,6 +201,44 @@ L.Map.SlideShow = L.Handler.extend({
...
+	_handleSlideWindowLoaded: function() {
+		const iframe = this._slideShowWindowProxy.document.querySelector('iframe');
+
+		if (iframe) {
+			iframe.contentWindow.focus();
+			iframe.contentWindow.addEventListener('keydown', this._onSlideWindowKeyPress.bind(this));
+		}
+	},
+
+	_generateSlideWindowHtml: function(title, slideURL) {
```
Comment (eszkadev) [inline browser/src/map/handler/Map.SlideShow.js:213]: we need to sanitize these variables to avoid injecting HTML

**Review by eszkadev — APPROVED** [review]

**Pull request merged**